### PR TITLE
chore(types): add type information

### PR DESCRIPTION
**Pull request by sendilkumarn**
**What kind of change does this PR introduce?**

Adds some type related information and removed where it is obvious

**Did you add tests for your changes?**
N/A
**If relevant, did you update the documentation?**
N/A
**Summary**
leverage on type inference

r? @evenstensberg @ematipico  @dhruvdutt 

**Does this PR introduce a breaking change?**

**Other information**

### CODE REVIEW

File: packages/generators/add-generator.ts [hunk 1]
```diff
@@ -73,7 +72,7 @@ const traverseAndGetProperties = (arr: object[], prop: string): boolean => {
 const searchProps = (answers: object, input: string): Promise<string[]> => {
 	input = input || "";
 	return Promise.resolve(
-		PROPS.filter((prop: string): boolean =>
+		PROPS.filter((prop): boolean =>
```
Comment (misterdev) [inline packages/generators/add-generator.ts]: In other occurrences of `.filter` you are keeping the type of the parameter. I would keep here too

File: packages/generators/add-generator.ts [hunk 2]
```diff
@@ -162,7 +161,7 @@ export default class AddGenerator extends Generator {
 							this.configuration.config.item = action;
 						});
 				}
-				const temp: string = action;
+				const temp = action;
```
Comment (misterdev) [inline packages/generators/add-generator.ts]: It doesn't look obvious to me. I would keep the type here

File: packages/init/init.ts [hunk 3]
```diff
@@ -42,11 +42,11 @@ export default function runTransform(webpackProperties: IWebpackProperties, acti
 			.keys(webpackProperties)
 			.filter((p: string): boolean => p !== "configFile" && p !== "configPath");
 
-	const initActionNotDefined: boolean = (action && action !== "init") || false;
+	const initActionNotDefined = (action && action !== "init") || false;
 
 	webpackConfig.forEach((scaffoldPiece: string): Promise<void> => {
 		const config: IConfiguration = webpackProperties[scaffoldPiece];
-		const transformations: string[] = mapOptionsToTransform(config);
+		const transformations = mapOptionsToTransform(config);
```
Comment (misterdev) [inline packages/init/init.ts]: It doesn't look obvious to me what's the return type of `mapOptionsToTransform`

File: packages/generators/add-generator.ts (hunk 1 above)
Comment (sendilkumarn) [reply packages/generators/add-generator.ts]: You are correct here. we need to have it here.

File: packages/init/init.ts (hunk 3 above)
Comment (sendilkumarn) [reply packages/init/init.ts]: My inspiration for this and the above one is from the Typescript core repo. Again here the type is obvious if you know the method signature or the variable and we don't need to be verbose.

But well, I agree it is highly subjective...

Comment (ematipico) [reply packages/init/init.ts]: Actually there's no need to assign a type if the function already has a signature (and just one, where it returns always a string).

File: packages/generators/add-generator.ts (hunk 2 above)
Comment (ematipico) [reply packages/generators/add-generator.ts]: `action` is already a string, so `temp` inherits its type

File: packages/utils/ast-utils.ts [hunk 4]
```diff
@@ -230,7 +237,7 @@ function addOrUpdateConfigObject(
 
 	if (propertyExists) {
 		rootNode.properties
-			.filter((path: INode) => path.key.name === configProperty)
+			.filter((path: INode): boolean => path.key.name === configProperty)
```
Comment (ematipico) [inline packages/utils/ast-utils.ts]: Is it needed the type here?

**Comment by evenstensberg** [comment]
Needs a quick rebase @sendilkumarn

**Comment by sendilkumarn** [comment]
Done rebasing, let us merge this once the builds are green 🎉

**Pull request merged**